gene/etl/ensembl.py

```diff
@@ -38,50 +44,85 @@ def __init__(
         :param src_data_dir: Data directory for Ensembl
         """
         super().__init__(database, host, data_dir, src_data_dir)
+        self._data_file_pattern = re.compile(r"ensembl_(GRCh\d+)_(\d+)\.gff3")
         self._sequence_location = SequenceLocation()
         self._version = None
-        self._fn = None
-        self._data_url = None
+        self._data_url = {}
         self._assembly = None
 
-    def _download_data(self) -> None:
-        """Download latest Ensembl GFF3 data file."""
+    def _is_up_to_date(self, data_file: Path) -> bool:
+        """Verify whether local data is up-to-date with latest available remote file.
+
+        :param data_file: path to latest local file
+        :return: True if data is up-to-date
+        :raise FileVersionError: if unable to parse version number from local file
+        :raise SourceFetchError: if unable to get latest version from remote source
+        """
+        local_match = re.match(self._data_file_pattern, data_file.name)
+        parse_msg = (
+            f"Unable to parse version number from local file: {data_file.absolute()}"
+        )
+        if not local_match or not local_match.groups():
+            raise GeneFileVersionError(parse_msg)
+        try:
+            version = int(local_match.groups()[1])
```

Comment:
Save as var so you don't have to call again
```suggestion
        local_match_groups = local_match.groups()
        if not local_match or not local_match_groups:
            raise GeneFileVersionError(parse_msg)
        try:
            version = int(local_match_groups[1])
```

Reply:
@korikuzma doesn't `local_match.groups()` raise an error if `local_match` is None?

Reply:
Oh right, wasn't thinking.